BUG: Cannot find `OpenBLAS` on Cygwin

### Describe your issue.

While I try to use `pip3 install maryam` to install `maryam` in Cygwin, it keeps throwing the following error:
`../scipy/meson.build:169:9: ERROR: Dependency "OpenBLAS" not found, tried pkgconfig and cmake`
because `maryam` depends on `scikit-learn` which depends on `scipy` ...

### Reproducing Code Example

Although I could reproduce the issue with `pip3 install maryam`, it is simpler to reproduce it with `pip3 install scipy` as follow.

<details>

```
yutta@SurfacePro9 /cygdrive/c/Users/yutta/eclipse-workspace/Maryam
$ pip3 list
Package                       Version
----------------------------- -------
attrs                         23.2.0
Babel                         2.12.1
build                         1.0.3
cffi                          1.16.0
chardet                       4.0.0
Cython                        0.29.33
docutils                      0.18.1
html5lib                      1.1
hypothesis                    6.35.0
idna                          3.3
imagesize                     1.3.0
iniconfig                     2.0.0
Jinja2                        3.1.2
lxml                          4.7.1
MarkupSafe                    2.1.2
meson                         1.2.3
nose3                         1.3.8
numpy                         1.21.4
olefile                       0.46
packaging                     21.3
Pillow                        8.4.0
pip                           24.0
pkgconfig                     1.5.5
platformdirs                  4.2.0
pluggy                        1.0.0
ply                           3.11
py                            1.11.0
pycparser                     2.21
Pygments                      2.10.0
pyparsing                     3.0.6
pytest                        7.3.0
python-dateutil               2.8.2
pytz                          2023.3
railroad-diagrams             3.0.1
requests                      2.31.0
setuptools                    69.2.0
six                           1.16.0
snowballstemmer               2.2.0
sortedcontainers              2.4.0
Sphinx                        4.4.0
sphinxcontrib-serializinghtml 1.1.5
toml                          0.10.2
urllib3                       1.26.7
webencodings                  0.5.1

yutta@SurfacePro9 /cygdrive/c/Users/yutta/eclipse-workspace/Maryam
$ pip3 install scipy
```

</details>

### Error message

```
      ../scipy/meson.build:163:9: ERROR: Dependency "OpenBLAS" not found, tried pkgconfig and cmake
```

Full output:

<details>

```
Collecting scipy
  Using cached scipy-1.13.0.tar.gz (57.2 MB)
  Installing build dependencies ... done
  Getting requirements to build wheel ... done
  Preparing metadata (pyproject.toml) ... error
  error: subprocess-exited-with-error

  × Preparing metadata (pyproject.toml) did not run successfully.
  │ exit code: 1
  ╰─> [46 lines of output]
      + meson setup /cygdrive/c/Users/yutta/AppData/Local/Temp/pip-install-mkkespja/scipy_16c06cfb07e24d2391b8c36c833c7d83 /cygdrive/c/Users/yutta/AppData/Local/Temp/pip-install-mkkespja/scipy_16c06cfb07e24d2391b8c36c833c7d83/.m
esonpy-y6vwx62o -Dbuildtype=release -Db_ndebug=if-release -Db_vscrt=md --native-file=/cygdrive/c/Users/yutta/AppData/Local/Temp/pip-install-mkkespja/scipy_16c06cfb07e24d2391b8c36c833c7d83/.mesonpy-y6vwx62o/meson-python-native-file.ini
      The Meson build system
      Version: 1.4.0
      Source dir: /cygdrive/c/Users/yutta/AppData/Local/Temp/pip-install-mkkespja/scipy_16c06cfb07e24d2391b8c36c833c7d83
      Build dir: /cygdrive/c/Users/yutta/AppData/Local/Temp/pip-install-mkkespja/scipy_16c06cfb07e24d2391b8c36c833c7d83/.mesonpy-y6vwx62o
      Build type: native build
      Project name: scipy
      Project version: 1.13.0
      C compiler for the host machine: gcc (gcc 11.4.0 "gcc (GCC) 11.4.0")
      C linker for the host machine: gcc ld.bfd 2.42
      C++ compiler for the host machine: c++ (gcc 11.4.0 "c++ (GCC) 11.4.0")
      C++ linker for the host machine: c++ ld.bfd 2.42
      Cython compiler for the host machine: cython (cython 3.0.10)
      Host machine cpu family: x86_64
      Host machine cpu: x86_64
      Program python found: YES (/usr/bin/python3.9.exe)
      Found pkg-config: YES (/usr/bin/pkg-config) 2.2.0
      Run-time dependency python found: YES 3.9
      Program cython found: YES (/cygdrive/c/Users/yutta/AppData/Local/Temp/pip-build-env-kikt09he/overlay/bin/cython)
      Compiler for C supports arguments -Wno-unused-but-set-variable: YES
      Compiler for C supports arguments -Wno-unused-function: YES
      Compiler for C supports arguments -Wno-conversion: YES
      Compiler for C supports arguments -Wno-misleading-indentation: YES
      Library m found: YES
      Fortran compiler for the host machine: gfortran (gcc 11.4.0 "GNU Fortran (GCC) 11.4.0")
      Fortran linker for the host machine: gfortran ld.bfd 2.42
      Compiler for Fortran supports arguments -Wno-conversion: YES
      Checking if "-Wl,--version-script" : links: YES
      Program tools/generate_f2pymod.py found: YES (/usr/bin/python3.9.exe /cygdrive/c/Users/yutta/AppData/Local/Temp/pip-install-mkkespja/scipy_16c06cfb07e24d2391b8c36c833c7d83/tools/generate_f2pymod.py)
      Program scipy/_build_utils/tempita.py found: YES (/usr/bin/python3.9.exe /cygdrive/c/Users/yutta/AppData/Local/Temp/pip-install-mkkespja/scipy_16c06cfb07e24d2391b8c36c833c7d83/scipy/_build_utils/tempita.py)
      Program pythran found: YES 0.15.0 0.15.0 (/cygdrive/c/Users/yutta/AppData/Local/Temp/pip-build-env-kikt09he/overlay/bin/pythran)
      Found CMake: /usr/bin/cmake (3.28.3)
      WARNING: CMake Toolchain: Failed to determine CMake compilers state
      Run-time dependency xsimd found: NO (tried pkgconfig and cmake)
      Run-time dependency threads found: YES
      Library npymath found: YES
      Library npyrandom found: YES
      pybind11-config found: YES (/cygdrive/c/Users/yutta/AppData/Local/Temp/pip-build-env-kikt09he/overlay/bin/pybind11-config) 2.12.0
      Run-time dependency pybind11 found: YES 2.12.0
      Run-time dependency scipy-openblas found: NO (tried pkgconfig)
      Run-time dependency openblas found: NO (tried pkgconfig and cmake)
      Run-time dependency openblas found: NO (tried pkgconfig and cmake)

      ../scipy/meson.build:163:9: ERROR: Dependency "OpenBLAS" not found, tried pkgconfig and cmake

      A full log can be found at /cygdrive/c/Users/yutta/AppData/Local/Temp/pip-install-mkkespja/scipy_16c06cfb07e24d2391b8c36c833c7d83/.mesonpy-y6vwx62o/meson-logs/meson-log.txt
      [end of output]

  note: This error originates from a subprocess, and is likely not a problem with pip.
error: metadata-generation-failed

× Encountered error while generating package metadata.
╰─> See above for output.

note: This is an issue with the package mentioned above, not pip.
hint: See above for details.
```

</details>

### SciPy/NumPy/Python version and system information

I install Cygwin in Windows 11 arm 64. For openblas, my Cygwin only has three packages: `libopenblas` , `openblas-debuginfo`, and `openblas-doc` with all version 0.3.25-1 . For Python, the `python39` package with version 3.9.16-1 install the following python and pip:
```
yutta@SurfacePro9 /cygdrive/c/Users/yutta/eclipse-workspace/Maryam
$ type python3.9
python3.9 is /usr/bin/python3.9

yutta@SurfacePro9 /cygdrive/c/Users/yutta/eclipse-workspace/Maryam
$ type pip3     
pip3 is hashed (/usr/bin/pip3)
```

I tried, but my source was directly downloaded by `pip3 install scipy -Csetup-args=-Dblas=blas -Csetup-args=-Dlapack=lapack`. I tried to find the location of `scipy-1.13.0.tar.gz` with 'pip cache', but `pip3 cache list` only gives me `whl` files. Also, `pip3 download scipy` gives me the original error: `../scipy/meson.build:163:9: ERROR: Dependency "OpenBLAS" not found, tried pkgconfig and cmake`... Anyway, I don't know how to bypass the building process to get the `scipy-1.13.0.tar.gz` downloaded by `pip3` directly...

Since I didn't find the `scipy-1.13.0.tar.gz` downloaded by `pip3`, I tried to check out the branch `https://github.com/scipy/scipy/tree/maintenance/0.13.x`, add `#include <Python.h>` at the top of `scipy/io/_fast_matrix_market/src/_fmm_core.cpp`, and tried again. No matter whether I add `#include <Python.h>`, I got the following same error message different from the previous one. I guess this scale up the problem and doesn't really help, but I don't know what's the next step if I could not get the `scipy-1.13.0.tar.gz` downloaded by `pip3`...
```
yutta@SurfacePro9 /cygdrive/c/Users/yutta/eclipse-workspace/scipy
$ pip3 install . -Csetup-args=-Dblas=blas -Csetup-args=-Dlapack=lapack
Processing /cygdrive/c/Users/yutta/eclipse-workspace/scipy
  Installing build dependencies ... done
  Getting requirements to build wheel ... done
  Preparing metadata (pyproject.toml) ... error
  error: subprocess-exited-with-error

  × Preparing metadata (pyproject.toml) did not run successfully.
  │ exit code: 1
  ╰─> [282 lines of output]
      + meson setup /cygdrive/c/Users/yutta/eclipse-workspace/scipy /cygdrive/c/Users/yutta/eclipse-workspa
ce/scipy/.mesonpy-c2vvzjdb -Dbuildtype=release -Db_ndebug=if-release -Db_vscrt=md -Dblas=blas -Dlapack=lapa
ck --native-file=/cygdrive/c/Users/yutta/eclipse-workspace/scipy/.mesonpy-c2vvzjdb/meson-python-native-file.ini
      The Meson build system
      Version: 1.4.0
      Source dir: /cygdrive/c/Users/yutta/eclipse-workspace/scipy
      Build dir: /cygdrive/c/Users/yutta/eclipse-workspace/scipy/.mesonpy-c2vvzjdb
      Build type: native build
      Project name: scipy
      Project version: 1.13.1.dev0
      C compiler for the host machine: gcc (gcc 11.4.0 "gcc (GCC) 11.4.0")
      C linker for the host machine: gcc ld.bfd 2.42
      C++ compiler for the host machine: c++ (gcc 11.4.0 "c++ (GCC) 11.4.0")
      C++ linker for the host machine: c++ ld.bfd 2.42
      Cython compiler for the host machine: cython (cython 3.0.10)
      Host machine cpu family: x86_64
      Host machine cpu: x86_64
      Program python found: YES (/usr/bin/python3.9.exe)
      Found pkg-config: YES (/usr/bin/pkg-config) 2.2.0
      Run-time dependency python found: YES 3.9
      Program cython found: YES (/cygdrive/c/Users/yutta/AppData/Local/Temp/pip-build-env-tz_g4mk8/overlay/bin/cython)
      Compiler for C supports arguments -Wno-unused-but-set-variable: YES
      Compiler for C supports arguments -Wno-unused-function: YES
      Compiler for C supports arguments -Wno-conversion: YES
      Compiler for C supports arguments -Wno-misleading-indentation: YES
      Library m found: YES
      Fortran compiler for the host machine: gfortran (gcc 11.4.0 "GNU Fortran (GCC) 11.4.0")
      Fortran linker for the host machine: gfortran ld.bfd 2.42
      Compiler for Fortran supports arguments -Wno-conversion: YES
      Checking if "-Wl,--version-script" : links: YES
      Program tools/generate_f2pymod.py found: YES (/cygdrive/c/Users/yutta/eclipse-workspace/scipy/tools/generate_f2pymod.py)
      Program scipy/_build_utils/tempita.py found: YES (/cygdrive/c/Users/yutta/eclipse-workspace/scipy/scipy/_build_utils/tempita.py)
      Program pythran found: YES 0.15.0 0.15.0 (/cygdrive/c/Users/yutta/AppData/Local/Temp/pip-build-env-tz_g4mk8/overlay/bin/pythran)
      Found CMake: /usr/bin/cmake (3.28.3)
      WARNING: CMake Toolchain: Failed to determine CMake compilers state
      Run-time dependency xsimd found: NO (tried pkgconfig and cmake)
      Run-time dependency threads found: YES
      Library npymath found: YES
      Library npyrandom found: YES
      pybind11-config found: YES (/cygdrive/c/Users/yutta/AppData/Local/Temp/pip-build-env-tz_g4mk8/overlay/bin/pybind11-config) 2.12.0
      Run-time dependency pybind11 found: YES 2.12.0
      Run-time dependency blas found: YES 3.12.0
      Run-time dependency cblas found: YES 3.12.0
      Run-time dependency lapack found: YES 3.12.0
      Compiler for C supports arguments -Wno-maybe-uninitialized: YES
      Compiler for C supports arguments -Wno-discarded-qualifiers: YES
      Compiler for C supports arguments -Wno-empty-body: YES
      Compiler for C supports arguments -Wno-implicit-function-declaration: YES
      Compiler for C supports arguments -Wno-parentheses: YES
      Compiler for C supports arguments -Wno-switch: YES
      Compiler for C supports arguments -Wno-unused-label: YES
      Compiler for C supports arguments -Wno-unused-result: YES
      Compiler for C supports arguments -Wno-unused-variable: YES
      Compiler for C++ supports arguments -Wno-cpp: YES
      Compiler for C++ supports arguments -Wno-deprecated-declarations: YES
      Compiler for C++ supports arguments -Wno-class-memaccess: YES
      Compiler for C++ supports arguments -Wno-format-truncation: YES
      Compiler for C++ supports arguments -Wno-non-virtual-dtor: YES
      Compiler for C++ supports arguments -Wno-sign-compare: YES
      Compiler for C++ supports arguments -Wno-switch: YES
      Compiler for C++ supports arguments -Wno-terminate: YES
      Compiler for C++ supports arguments -Wno-unused-but-set-variable: YES
      Compiler for C++ supports arguments -Wno-unused-function: YES
      Compiler for C++ supports arguments -Wno-unused-local-typedefs: YES
      Compiler for C++ supports arguments -Wno-unused-variable: YES
      Compiler for C++ supports arguments -Wno-int-in-bool-context: YES
      Compiler for Fortran supports arguments -Wno-argument-mismatch: YES
      Compiler for Fortran supports arguments -Wno-conversion: YES (cached)
      Compiler for Fortran supports arguments -Wno-intrinsic-shadow: YES
      Compiler for Fortran supports arguments -Wno-maybe-uninitialized: YES
      Compiler for Fortran supports arguments -Wno-surprising: YES
      Compiler for Fortran supports arguments -Wno-uninitialized: YES
      Compiler for Fortran supports arguments -Wno-unused-dummy-argument: YES
      Compiler for Fortran supports arguments -Wno-unused-label: YES
      Compiler for Fortran supports arguments -Wno-unused-variable: YES
      Compiler for Fortran supports arguments -Wno-tabs: YES
      Compiler for Fortran supports arguments -Wno-argument-mismatch: YES (cached)
      Compiler for Fortran supports arguments -Wno-conversion: YES (cached)
      Compiler for Fortran supports arguments -Wno-maybe-uninitialized: YES (cached)
      Compiler for Fortran supports arguments -Wno-unused-dummy-argument: YES (cached)
      Compiler for Fortran supports arguments -Wno-unused-label: YES (cached)
      Compiler for Fortran supports arguments -Wno-unused-variable: YES (cached)
      Compiler for Fortran supports arguments -Wno-tabs: YES (cached)
      Checking if "Check atomic builtins without -latomic" : links: YES
      Configuring __config__.py using configuration
      Checking for function "open_memstream" : NO
      Configuring messagestream_config.h using configuration
      Compiler for Fortran supports arguments -w: YES
      Checking for size of "void*" : 8
      Compiler for Fortran supports arguments -w: YES (cached)
      Build targets in project: 234

      scipy 1.13.1.dev0

        User defined options
          Native files: /cygdrive/c/Users/yutta/eclipse-workspace/scipy/.mesonpy-c2vvzjdb/meson-python-native-file.ini
          buildtype   : release
          b_ndebug    : if-release
          b_vscrt     : md
          blas        : blas
          lapack      : lapack

      Found ninja-1.11.1 at /usr/bin/ninja
      + /usr/bin/ninja
      [1/1475] Copying file scipy/__init__.py
      [2/1475] Generating scipy/generate-version with a custom command
      [3/1475] Copying file scipy/linalg.pxd
      [4/1475] Copying file scipy/optimize.pxd
      [5/1475] Copying file scipy/special.pxd
      [6/1475] Copying file scipy/_lib/__init__.py
      [7/1475] Copying file scipy/_lib/_ccallback_c.pxd
      [8/1475] Copying file scipy/_lib/ccallback.pxd
      [9/1475] Copying file scipy/_lib/messagestream.pxd
      [10/1475] Compiling C object scipy/libdummy_g77_abi_wrappers.a.p/_build_utils_src_wrap_dummy_g77_abi.c.o
      [11/1475] Copying file scipy/special/__init__.py
      [12/1475] Linking static target scipy/libdummy_g77_abi_wrappers.a
      [13/1475] Compiling C object scipy/_lib/_fpumode.cpython-39-x86_64-cygwin.dll.p/_fpumode.c.o
      [14/1475] Copying file scipy/special/_agm.pxd
      [15/1475] Linking target scipy/_lib/_fpumode.cpython-39-x86_64-cygwin.dll
      [16/1475] Compiling C object scipy/_lib/_test_ccallback.cpython-39-x86_64-cygwin.dll.p/src__test_ccallback.c.o
      [17/1475] Copying file scipy/special/_boxcox.pxd
      [18/1475] Copying file scipy/special/_cdflib.pxd
      [19/1475] Linking target scipy/_lib/_test_ccallback.cpython-39-x86_64-cygwin.dll
      [20/1475] Compiling C++ object scipy/_lib/_uarray/_uarray.cpython-39-x86_64-cygwin.dll.p/vectorcall.cxx.o
      [21/1475] Generating 'scipy/_lib/_test_deprecation_call.cpython-39-x86_64-cygwin.dll.p/_test_deprecation_call.c'
      [22/1475] Generating 'scipy/_lib/_test_deprecation_def.cpython-39-x86_64-cygwin.dll.p/_test_deprecation_def.c'
      [23/1475] Copying file scipy/special/_cdflib_wrappers.pxd
      [24/1475] Copying file scipy/special/_cephes.pxd
      [25/1475] Copying file scipy/special/_complexstuff.pxd
      [26/1475] Copying file scipy/special/_convex_analysis.pxd
      [27/1475] Copying file scipy/special/_cunity.pxd
      [28/1475] Copying file scipy/special/_digamma.pxd
      [29/1475] Copying file scipy/special/_ellip_harm.pxd
      [30/1475] Copying file scipy/special/_ellip_harm_2.pxd
      [31/1475] Generating 'scipy/_lib/_ccallback_c.cpython-39-x86_64-cygwin.dll.p/_ccallback_c.c'
      [32/1475] Copying file scipy/special/_ellipk.pxd
      [33/1475] Copying file scipy/special/_exprel.pxd
      [34/1475] Copying file scipy/special/_factorial.pxd
      [35/1475] Compiling C object scipy/lib_fortranobject.a.p/dece4e8a540d38c86625681cea6b00dc7d57a7f3_site-packages_numpy_f2py_src_fortranobject.c.o
      [36/1475] Copying file scipy/special/_hyp0f1.pxd
      [37/1475] Linking static target scipy/lib_fortranobject.a
      [38/1475] Copying file scipy/special/_hyp2f1.pxd
      [39/1475] Copying file scipy/special/_hypergeometric.pxd
      [40/1475] Copying file scipy/special/_legacy.pxd
      [41/1475] Copying file scipy/special/_ndtri_exp.pxd
      [42/1475] Copying file scipy/special/_sici.pxd
      [43/1475] Copying file scipy/special/_spence.pxd
      [44/1475] Copying file scipy/special/_spherical_bessel.pxd
      [45/1475] Copying file scipy/special/_trig.pxd
      [46/1475] Copying file scipy/special/_wright_bessel.pxd
      [47/1475] Copying file scipy/special/_xlogy.pxd
      [48/1475] Copying file scipy/special/orthogonal_eval.pxd
      [49/1475] Copying file scipy/special/sf_error.pxd
      [50/1475] Generating 'scipy/_lib/messagestream.cpython-39-x86_64-cygwin.dll.p/messagestream.c'       
      [51/1475] Copying file scipy/special/sph_harm.pxd
      [52/1475] Copying file scipy/special/_cython_special.pxi
      [53/1475] Copying file scipy/special/_cython_special_custom.pxi
      [54/1475] Copying file scipy/special/_ufuncs_extra_code.pxi
      [55/1475] Copying file scipy/special/_ufuncs_extra_code_common.pxi
      [56/1475] Compiling C object scipy/special/libcephes.a.p/cephes_bdtr.c.o
      [57/1475] Compiling C object scipy/special/libcephes.a.p/cephes_btdtr.c.o
      [58/1475] Compiling C object scipy/_lib/_test_deprecation_def.cpython-39-x86_64-cygwin.dll.p/meson-generated__test_deprecation_def.c.o
      [59/1475] Compiling C object scipy/special/libcephes.a.p/cephes_airy.c.o
      [60/1475] Compiling C object scipy/special/libcephes.a.p/cephes_besselpoly.c.o
      [61/1475] Compiling C object scipy/special/libcephes.a.p/cephes_beta.c.o
      [62/1475] Compiling C object scipy/_lib/_test_deprecation_call.cpython-39-x86_64-cygwin.dll.p/meson-generated__test_deprecation_call.c.o
      [63/1475] Linking target scipy/_lib/_test_deprecation_def.cpython-39-x86_64-cygwin.dll
      [64/1475] Compiling C object scipy/special/libcephes.a.p/cephes_cbrt.c.o
      [65/1475] Compiling C object scipy/special/libcephes.a.p/cephes_chbevl.c.o
      [66/1475] Compiling C object scipy/special/libcephes.a.p/cephes_chdtr.c.o
      [67/1475] Compiling C object scipy/special/libcephes.a.p/cephes_const.c.o
      [68/1475] Compiling C object scipy/special/libcephes.a.p/cephes_dawsn.c.o
      [69/1475] Linking target scipy/_lib/_test_deprecation_call.cpython-39-x86_64-cygwin.dll
      [70/1475] Compiling C object scipy/special/libcephes.a.p/cephes_ellpe.c.o
      [71/1475] Compiling C object scipy/special/libcephes.a.p/cephes_ellik.c.o
      [72/1475] Compiling C object scipy/special/libcephes.a.p/cephes_ellie.c.o
      [73/1475] Compiling C object scipy/special/libcephes.a.p/cephes_ellpj.c.o
      [74/1475] Compiling C object scipy/special/libcephes.a.p/cephes_ellpk.c.o
      [75/1475] Compiling C object scipy/special/libcephes.a.p/cephes_erfinv.c.o
      [76/1475] Compiling C object scipy/special/libcephes.a.p/cephes_exp10.c.o
      [77/1475] Compiling C object scipy/special/libcephes.a.p/cephes_exp2.c.o
      [78/1475] Compiling C object scipy/special/libcephes.a.p/cephes_fdtr.c.o
      [79/1475] Compiling C object scipy/special/libcephes.a.p/cephes_expn.c.o
      [80/1475] Compiling C object scipy/special/libcephes.a.p/cephes_fresnl.c.o
      [81/1475] Compiling C object scipy/special/libcephes.a.p/cephes_gamma.c.o
      [82/1475] Compiling C object scipy/special/libcephes.a.p/cephes_gammasgn.c.o
      [83/1475] Compiling C object scipy/special/libcephes.a.p/cephes_gdtr.c.o
      [84/1475] Compiling C object scipy/special/libcephes.a.p/cephes_i0.c.o
      [85/1475] Compiling C object scipy/special/libcephes.a.p/cephes_hyperg.c.o
      [86/1475] Compiling C object scipy/special/libcephes.a.p/cephes_dd_real.c.o
      [87/1475] Compiling C object scipy/special/libcephes.a.p/cephes_i1.c.o
      [88/1475] Compiling C object scipy/special/libcephes.a.p/cephes_hyp2f1.c.o
      [89/1475] Compiling C object scipy/special/libcephes.a.p/cephes_igami.c.o
      [90/1475] Compiling C object scipy/special/libcephes.a.p/cephes_igam.c.o
      [91/1475] Compiling C object scipy/special/libcephes.a.p/cephes_incbet.c.o
      [92/1475] Compiling C object scipy/special/libcephes.a.p/cephes_incbi.c.o
      [93/1475] Compiling C object scipy/special/libcephes.a.p/cephes_j0.c.o
      [94/1475] Compiling C object scipy/special/libcephes.a.p/cephes_j1.c.o
      [95/1475] Compiling C object scipy/special/libcephes.a.p/cephes_k1.c.o
      [96/1475] Compiling C object scipy/special/libcephes.a.p/cephes_k0.c.o
      [97/1475] Compiling C object scipy/special/libcephes.a.p/cephes_kn.c.o
      [98/1475] Compiling C object scipy/special/libcephes.a.p/cephes_lanczos.c.o
      [99/1475] Compiling C object scipy/special/libcephes.a.p/cephes_jv.c.o
      [100/1475] Compiling C object scipy/special/libcephes.a.p/cephes_nbdtr.c.o
      [101/1475] Compiling C object scipy/special/libcephes.a.p/cephes_ndtri.c.o
      [102/1475] Compiling C object scipy/special/libcephes.a.p/cephes_ndtr.c.o
      [103/1475] Compiling C object scipy/special/libcephes.a.p/cephes_pdtr.c.o
      [104/1475] Compiling C object scipy/special/libcephes.a.p/cephes_owens_t.c.o
      [105/1475] Compiling C object scipy/special/libcephes.a.p/cephes_poch.c.o
      [106/1475] Compiling C object scipy/special/libcephes.a.p/cephes_psi.c.o
      [107/1475] Compiling C object scipy/special/libcephes.a.p/cephes_round.c.o
      [108/1475] Compiling C object scipy/special/libcephes.a.p/cephes_rgamma.c.o
      [109/1475] Compiling C object scipy/special/libcephes.a.p/cephes_shichi.c.o
      [110/1475] Compiling C object scipy/special/libcephes.a.p/cephes_sici.c.o
      [111/1475] Compiling C object scipy/special/libcephes.a.p/cephes_scipy_iv.c.o
      [112/1475] Compiling C object scipy/special/libcephes.a.p/cephes_kolmogorov.c.o
      [113/1475] Compiling C object scipy/special/libcephes.a.p/cephes_sinpi.c.o
      [114/1475] Compiling C object scipy/special/libcephes.a.p/cephes_sindg.c.o
      [115/1475] Compiling C object scipy/special/libcephes.a.p/cephes_spence.c.o
      [116/1475] Compiling C object scipy/special/libcephes.a.p/cephes_stdtr.c.o
      [117/1475] Compiling C object scipy/special/libcephes.a.p/cephes_tandg.c.o
      [118/1475] Compiling C object scipy/special/libcephes.a.p/cephes_tukey.c.o
      [119/1475] Compiling C object scipy/special/libcephes.a.p/cephes_yn.c.o
      [120/1475] Compiling C object scipy/special/libcephes.a.p/cephes_yv.c.o
      [121/1475] Compiling C object scipy/special/libcephes.a.p/cephes_unity.c.o
      [122/1475] Compiling C object scipy/special/libcephes.a.p/cephes_zeta.c.o
      [123/1475] Compiling C object scipy/special/_ufuncs.cpython-39-x86_64-cygwin.dll.p/_cosine.c.o       
      [124/1475] Generating scipy/special/cython_special with a custom command
      [125/1475] Compiling C object scipy/special/libcephes.a.p/cephes_zetac.c.o
      [126/1475] Compiling C object scipy/_lib/messagestream.cpython-39-x86_64-cygwin.dll.p/meson-generated_messagestream.c.o
      [127/1475] Compiling C object scipy/special/_ufuncs.cpython-39-x86_64-cygwin.dll.p/scaled_exp1.c.o   
      [128/1475] Compiling C object scipy/special/libcephes.a.p/cephes_struve.c.o
      [129/1475] Generating scipy/special/_dummy_cython_special.pxd with a custom command
      [130/1475] Linking static target scipy/special/libcephes.a
      [131/1475] Linking target scipy/_lib/messagestream.cpython-39-x86_64-cygwin.dll
      [132/1475] Compiling C object scipy/special/_ufuncs.cpython-39-x86_64-cygwin.dll.p/sf_error.c.o      
      [133/1475] Compiling C object scipy/_lib/_ccallback_c.cpython-39-x86_64-cygwin.dll.p/meson-generated__ccallback_c.c.o
      [134/1475] Generating 'scipy/special/_specfun.cpython-39-x86_64-cygwin.dll.p/_specfun.cpp'
      [135/1475] Linking target scipy/_lib/_ccallback_c.cpython-39-x86_64-cygwin.dll
      [136/1475] Compiling C++ object scipy/_lib/_uarray/_uarray.cpython-39-x86_64-cygwin.dll.p/_uarray_dispatch.cxx.o
      [137/1475] Linking target scipy/_lib/_uarray/_uarray.cpython-39-x86_64-cygwin.dll
      [138/1475] Generating 'scipy/special/_ufuncs_cxx.cpython-39-x86_64-cygwin.dll.p/_ufuncs_cxx.cpp'     
      [139/1475] Compiling C++ object scipy/special/_ufuncs_cxx.cpython-39-x86_64-cygwin.dll.p/_faddeeva.cxx.o
      [140/1475] Compiling C++ object scipy/special/_ufuncs_cxx.cpython-39-x86_64-cygwin.dll.p/_wright.cxx.o
      [141/1475] Compiling C++ object scipy/special/_ufuncs_cxx.cpython-39-x86_64-cygwin.dll.p/sf_error.cc.o
      [142/1475] Generating 'scipy/special/_cdflib.cpython-39-x86_64-cygwin.dll.p/_cdflib.c'
      [143/1475] Compiling C++ object scipy/special/_ufuncs_cxx.cpython-39-x86_64-cygwin.dll.p/wright.cc.o 
      [144/1475] Compiling C++ object scipy/special/_ufuncs_cxx.cpython-39-x86_64-cygwin.dll.p/Faddeeva.cc.o
      [145/1475] Compiling C object scipy/special/_ellip_harm_2.cpython-39-x86_64-cygwin.dll.p/sf_error.c.o
      [146/1475] Generating 'scipy/special/_ellip_harm_2.cpython-39-x86_64-cygwin.dll.p/_ellip_harm_2.c'   
      [147/1475] Compiling C object scipy/special/cython_special.cpython-39-x86_64-cygwin.dll.p/_cosine.c.o
      [148/1475] Compiling C object scipy/special/cython_special.cpython-39-x86_64-cygwin.dll.p/scaled_exp1.c.o
      [149/1475] Compiling C++ object scipy/special/_ufuncs.cpython-39-x86_64-cygwin.dll.p/amos_wrappers.cpp.o
      [150/1475] Compiling C++ object scipy/special/_ufuncs_cxx.cpython-39-x86_64-cygwin.dll.p/ellint_carlson_wrap.cxx.o
      [151/1475] Generating 'scipy/special/_ufuncs.cpython-39-x86_64-cygwin.dll.p/_ufuncs.c'
      [152/1475] Compiling C object scipy/special/cython_special.cpython-39-x86_64-cygwin.dll.p/sf_error.c.o
      [153/1475] Compiling C object scipy/special/_ellip_harm_2.cpython-39-x86_64-cygwin.dll.p/meson-generated__ellip_harm_2.c.o
      [154/1475] Compiling C++ object scipy/special/_ufuncs.cpython-39-x86_64-cygwin.dll.p/specfun_wrappers.cpp.o
      [155/1475] Linking target scipy/special/_ellip_harm_2.cpython-39-x86_64-cygwin.dll
      [156/1475] Generating 'scipy/special/_comb.cpython-39-x86_64-cygwin.dll.p/_comb.c'
      [157/1475] Compiling C object scipy/special/_cdflib.cpython-39-x86_64-cygwin.dll.p/meson-generated__cdflib.c.o
      [158/1475] Linking target scipy/special/_cdflib.cpython-39-x86_64-cygwin.dll
      [159/1475] Generating 'scipy/special/cython_special.cpython-39-x86_64-cygwin.dll.p/cython_special.c' 
      [160/1475] Compiling C object scipy/special/_test_internal.cpython-39-x86_64-cygwin.dll.p/cephes_dd_real.c.o
      [161/1475] Generating 'scipy/special/_test_internal.cpython-39-x86_64-cygwin.dll.p/_test_internal.c' 
      [162/1475] Compiling C++ object scipy/special/cython_special.cpython-39-x86_64-cygwin.dll.p/amos_wrappers.cpp.o
      [163/1475] Generating scipy/special/_data_gsl with a custom command
      [164/1475] Generating scipy/special/_data_boost with a custom command
      [165/1475] Copying file scipy/linalg/__init__.py
      [166/1475] Generating scipy/special/_data_local with a custom command
      [167/1475] Generating scipy/linalg/fblas_module with a custom command
      FAILED: scipy/linalg/_fblasmodule.c
      /cygdrive/c/Users/yutta/eclipse-workspace/scipy/tools/generate_f2pymod.py ../scipy/linalg/fblas.pyf.src -o scipy/linalg
      /usr/bin/env: 'python3': No such file or directory
      [168/1475] Copying file scipy/linalg/_cythonized_array_utils.pxd
      [169/1475] Generating scipy/linalg/cython_linalg with a custom command (wrapped by meson to set env) 
      [170/1475] Compiling C object scipy/special/_comb.cpython-39-x86_64-cygwin.dll.p/meson-generated__comb.c.o
      [171/1475] Compiling C++ object scipy/special/_specfun.cpython-39-x86_64-cygwin.dll.p/meson-generated__specfun.cpp.o
      [172/1475] Compiling C++ object scipy/special/_ufuncs_cxx.cpython-39-x86_64-cygwin.dll.p/meson-generated__ufuncs_cxx.cpp.o
      [173/1475] Compiling C++ object scipy/special/cython_special.cpython-39-x86_64-cygwin.dll.p/specfun_wrappers.cpp.o
      [174/1475] Compiling C object scipy/special/_test_internal.cpython-39-x86_64-cygwin.dll.p/meson-generated__test_internal.c.o
      [175/1475] Compiling C object scipy/special/_ufuncs.cpython-39-x86_64-cygwin.dll.p/meson-generated__ufuncs.c.o
      [176/1475] Compiling C object scipy/special/cython_special.cpython-39-x86_64-cygwin.dll.p/meson-generated_cython_special.c.o
      ninja: build stopped: subcommand failed.
      [end of output]

  note: This error originates from a subprocess, and is likely not a problem with pip.
error: metadata-generation-failed

× Encountered error while generating package metadata.
╰─> See above for output.

note: This is an issue with the package mentioned above, not pip.
hint: See above for details.

yutta@SurfacePro9 /cygdrive/c/Users/yutta/eclipse-workspace/scipy
$ git status
Refresh index: 100% (3172/3172), done.
On branch maintenance/1.13.x
Your branch is up to date with 'origin/maintenance/1.13.x'.

Changes not staged for commit:
  (use "git add <file>..." to update what will be committed)
  (use "git restore <file>..." to discard changes in working directory)
        modified:   scipy/_lib/array_api_compat (new commits)
        modified:   scipy/_lib/boost_math (new commits)
        modified:   scipy/_lib/pocketfft (new commits)
        modified:   scipy/io/_fast_matrix_market/src/_fmm_core.cpp

no changes added to commit (use "git add" and/or "git commit -a")

yutta@SurfacePro9 /cygdrive/c/Users/yutta/eclipse-workspace/scipy
$ git diff
warning: in the working copy of 'scipy/io/_fast_matrix_market/src/_fmm_core.cpp', CRLF will be replaced by LF the next time Git touches it
diff --git a/scipy/_lib/array_api_compat b/scipy/_lib/array_api_compat
index 05548f028..fd22a73e1 160000
--- a/scipy/_lib/array_api_compat
+++ b/scipy/_lib/array_api_compat
-Subproject commit 05548f0281f13e7cb6a293a422ef8c1a90021413
+Subproject commit fd22a73e19b2273035e15e8bce57f767dee8dff7
diff --git a/scipy/_lib/boost_math b/scipy/_lib/boost_math
index 1a7be5d89..15c40fae1 160000
--- a/scipy/_lib/boost_math
+++ b/scipy/_lib/boost_math
@@ -1 +1 @@
-Subproject commit 1a7be5d895d266a870af7a6ed258e5bcf9838277
+Subproject commit 15c40fae1610fd83728d75e96d7377fe55f976f7
diff --git a/scipy/_lib/pocketfft b/scipy/_lib/pocketfft
index 0bf2b5122..936714274 160000
--- a/scipy/_lib/pocketfft
+++ b/scipy/_lib/pocketfft
@@ -1 +1 @@
-Subproject commit 0bf2b51224e7c4e973724bac67d2577c7831e27c
+Subproject commit 9367142748fcc9696a1c9e5a99b76ed9897c9daa
diff --git a/scipy/io/_fast_matrix_market/src/_fmm_core.cpp b/scipy/io/_fast_matrix_market/src/_fmm_core.cpp
index 86346d416..2a6bf05bc 100644
--- a/scipy/io/_fast_matrix_market/src/_fmm_core.cpp
:
-Subproject commit 05548f0281f13e7cb6a293a422ef8c1a90021413
+Subproject commit fd22a73e19b2273035e15e8bce57f767dee8dff7
diff --git a/scipy/_lib/boost_math b/scipy/_lib/boost_math
index 1a7be5d89..15c40fae1 160000
--- a/scipy/_lib/boost_math
+++ b/scipy/_lib/boost_math
@@ -1 +1 @@
-Subproject commit 1a7be5d895d266a870af7a6ed258e5bcf9838277
+Subproject commit 15c40fae1610fd83728d75e96d7377fe55f976f7
diff --git a/scipy/_lib/pocketfft b/scipy/_lib/pocketfft
index 0bf2b5122..936714274 160000
--- a/scipy/_lib/pocketfft
+++ b/scipy/_lib/pocketfft
@@ -1 +1 @@
-Subproject commit 0bf2b51224e7c4e973724bac67d2577c7831e27c
+Subproject commit 9367142748fcc9696a1c9e5a99b76ed9897c9daa
diff --git a/scipy/io/_fast_matrix_market/src/_fmm_core.cpp b/scipy/io/_fast_matrix_market/src/_fmm_core.cpp
index 86346d416..2a6bf05bc 100644
--- a/scipy/io/_fast_matrix_market/src/_fmm_core.cpp
+++ b/scipy/io/_fast_matrix_market/src/_fmm_core.cpp
@@ -2,6 +2,7 @@
 // Use of this source code is governed by the BSD 2-clause license found in the LICENSE.txt file.
 // SPDX-License-Identifier: BSD-2-Clause

+#include <Python.h>
 #include <fast_matrix_market/types.hpp>
 #include <cstdint>
 namespace fast_matrix_market {
(END)
+Subproject commit 15c40fae1610fd83728d75e96d7377fe55f976f7
diff --git a/scipy/_lib/pocketfft b/scipy/_lib/pocketfft
index 0bf2b5122..936714274 160000
--- a/scipy/_lib/pocketfft
+++ b/scipy/_lib/pocketfft
@@ -1 +1 @@
-Subproject commit 0bf2b51224e7c4e973724bac67d2577c7831e27c
+Subproject commit 9367142748fcc9696a1c9e5a99b76ed9897c9daa
diff --git a/scipy/io/_fast_matrix_market/src/_fmm_core.cpp b/scipy/io/_fast_matrix_market/src/_fmm_core.cpp
index 86346d416..2a6bf05bc 100644
--- a/scipy/io/_fast_matrix_market/src/_fmm_core.cpp
+++ b/scipy/io/_fast_matrix_market/src/_fmm_core.cpp
@@ -2,6 +2,7 @@
 // Use of this source code is governed by the BSD 2-clause license found in the LICENSE.txt file.
 // SPDX-License-Identifier: BSD-2-Clause
 
+#include <Python.h>
 #include <fast_matrix_market/types.hpp>
 #include <cstdint>
 namespace fast_matrix_market {


yutta@SurfacePro9 /cygdrive/c/Users/yutta/eclipse-workspace/scipy
$ git diff
warning: in the working copy of 'scipy/io/_fast_matrix_market/src/_fmm_core.cpp', CRLF will be replaced by LF the next time Git touches it
diff --git a/scipy/_lib/array_api_compat b/scipy/_lib/array_api_compat
index 05548f028..fd22a73e1 160000
--- a/scipy/_lib/array_api_compat
+++ b/scipy/_lib/array_api_compat
@@ -1 +1 @@
-Subproject commit 05548f0281f13e7cb6a293a422ef8c1a90021413
+Subproject commit fd22a73e19b2273035e15e8bce57f767dee8dff7
diff --git a/scipy/_lib/boost_math b/scipy/_lib/boost_math
index 1a7be5d89..15c40fae1 160000
--- a/scipy/_lib/boost_math
+++ b/scipy/_lib/boost_math
@@ -1 +1 @@
-Subproject commit 1a7be5d895d266a870af7a6ed258e5bcf9838277
+Subproject commit 15c40fae1610fd83728d75e96d7377fe55f976f7
diff --git a/scipy/_lib/pocketfft b/scipy/_lib/pocketfft
index 0bf2b5122..936714274 160000
--- a/scipy/_lib/pocketfft
+++ b/scipy/_lib/pocketfft
@@ -1 +1 @@

yutta@SurfacePro9 /cygdrive/c/Users/yutta/eclipse-workspace/scipy
$         
```

> This is what I saw when download `https://files.pythonhosted.org/packages/30/85/cdbf2c3c460fe5aae812917866392068a88d02f07de0fe31ce738734c477/scipy-1.12.0.tar.gz`, unzip to the current folder, adding `#include <Python.h>` in `scipy/io/_fast_matrix_market/src/_fmm_core.hpp`. Supprisingly, it doesn't work???
> 
> ```
>       [1007/1610] Compiling C++ object scipy/io/_fast_matrix_market/_fmm_core.cpython-39-x86_64-cygwin.dll.p/src__fmm_core_read_array.cpp.o
>       FAILED: scipy/io/_fast_matrix_market/_fmm_core.cpython-39-x86_64-cygwin.dll.p/src__fmm_core_read_array.cpp.o
>       c++ -Iscipy/io/_fast_matrix_market/_fmm_core.cpython-39-x86_64-cygwin.dll.p -Iscipy/io/_fast_matrix_market \
> ... -DFMM_SCIPY_PRUNE -DFMM_USE_FAST_FLOAT -DFMM_USE_RYU -DFMM_FROM_CHARS_INT_SUPPORTED -DFMM_TO_CHARS_INT_SUPPORTED -o scipy/io/_fast_matrix_market/_fmm_core.cpython-39-x86_64-cygwin.dll.p/src__fmm_core_read_array.cpp.o -c ../scipy/io/_fast_matrix_market/src/_fmm_core_read_array.cpp  
>       In file included from ../scipy/io/_fast_matrix_market/src/_fmm_core.hpp:13,
>                        from ../scipy/io/_fast_matrix_market/src/_fmm_core_read_array.cpp:5:
>       /cygdrive/c/Users/yutta/AppData/Local/Temp/pip-build-env-56cpe7_a/overlay/lib/python3.9/site-packages/pybind11/include/pybind11/pybind11.h: In member function 'char* pybind11::cpp_function::strdup_guard::operator()(const char*)':
>       /cygdrive/c/Users/yutta/AppData/Local/Temp/pip-build-env-56cpe7_a/overlay/lib/python3.9/site-packages/pybind11/include/pybind11/pybind11.h:76:36: error: 'strdup' was not declared in this scope; did you mean 'strcmp'?      
>          76 | #    define PYBIND11_COMPAT_STRDUP strdup
>             |                                    ^~~~~~
> ```

That rather strongly suggests that `_fmm_core_read_array.cpp` or `_fmm_core.hpp` have a system header included before `Python.h`.



> `pip3 install git+https://github.com/DWesl/scipy@patch-5 -Csetup-args=-Dblas=blas -Csetup-args=-Dlapack=lapack` works... Besides, if I want to further solve my original question, i.e., installing `scikit-learn` and `maryam`, do you know how I should start?

You should have a wheel now, which means `pip install maryam` should work fine.  I don't remember needing any modifications to scikit-learn last time I installed it.  I don't think SciPy would be a build dependency, so it might work with SciPy installed but not a wheel.  

If not, check `pyproject.toml` (or `setup.cfg` or `setup.py`) for `build_requires` or similar and change the scipy requirement to one of the locations that worked.

--- a/scipy/io/_fast_matrix_market/src/_fmm_core.cpp
+++ b/scipy/io/_fast_matrix_market/src/_fmm_core.cpp
@@ -2,6 +2,7 @@
 // Use of this source code is governed by the BSD 2-clause license found in the LICENSE.txt file.
 // SPDX-License-Identifier: BSD-2-Clause
 
+#include <Python.h>
 #include <fast_matrix_market/types.hpp>
 #include <cstdint>
 namespace fast_matrix_market {

--- a/scipy/special/_special_ufuncs.cpp
+++ b/scipy/special/_special_ufuncs.cpp
@@ -1,4 +1,7 @@
+#include "ufunc.h"
+
 #include <cmath>
+#include <complex>
 
 #include "ufunc.h"
 

--- a/scipy/special/_special_ufuncs.cpp
+++ b/scipy/special/_special_ufuncs.cpp
@@ -3,8 +3,6 @@
 #include <cmath>
 #include <complex>
 
-#include "ufunc.h"
-
 #include "sf_error.h"
 #include "special.h"
 #include "special/airy.h"

--- a/scipy/special/_wright.cxx
+++ b/scipy/special/_wright.cxx
@@ -1,6 +1,6 @@
-#include <complex>
-
 #include "_wright.h"
+
+#include <complex>
 
 using namespace std;
 

--- a/scipy/special/ufunc.h
+++ b/scipy/special/ufunc.h
@@ -1,3 +1,5 @@
+#pragma once
+
 #define PY_SSIZE_T_CLEAN
 #include <Python.h>
 

